bayesian/utility/db_gateway.py

```diff
@@ -98,6 +108,80 @@ def get_batch_ca_data(cls, ecosystem: str, packages) -> dict:
         logger.info("It took %s to fetch results from Gremlin.", elapsed_time)
         return response.json()
 
+    @classmethod
+    def get_batch_ca_data_for_pseudo_version(cls, ecosystem: str, packages) -> dict:
+        """Component analyses batch call only for pseudo version applicable for golang."""
+        logger.debug('Executing get_batch_ca_data_for_pseudo_version')
+
+        filter_packages = set()
+        package_version_map = {}
+        for pckg in packages:
+            package_name = pckg['name'].split('@')[0]
+            filter_packages.append(package_name)
+            package_version_map[package_name] = pckg["version"]
+
+        payload = {
+            'gremlin': cls.ca_batch_vuln_query,
+            'bindings': {
+                'ecosystem': ecosystem,
+                'packages': list(filter_packages)
+            }
+        }
+        started_at = time.time()
+        vuln_response = post(url=gremlin_url, data=json.dumps(payload))
```

Comment:
use try catch here

Reply:
next statement, raise_for_status() takes care of HTTP exception of 4XX and 5XX errors.